src/NHPersistenceCompatibilityTests/SharedDefinitions/ObjectFetcher.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+
+
+static class ObjectFetcher
+{
+    public static void Traverse(object instance, Type instanceType)
+    {
+        foreach (var propertyInfo in instanceType.GetProperties())
+        {
+            if (ReferenceEquals(propertyInfo.DeclaringType, typeof(object)))
```

Comment:
`A Type object that represents a type is unique; that is, two Type object references refer to the same object if and only if they represent the same type`. More details [here] (https://msdn.microsoft.com/en-us/library/system.type(v=vs.110).aspx) in `Comparing type objects for equality` section